virttest/utils_libvirtd.py

```diff
@@ -29,24 +29,34 @@ class Libvirtd(object):
     Class to manage libvirtd service on host or guest.
     """
 
-    def __init__(self, session=None):
+    def __init__(self, service_name=LIBVIRTD, session=None):
         """
         Initialize an service object for libvirtd.
 
+        :params service_name: Service name such as virtqemud or libvirtd.
         :params session: An session to guest or remote host.
         """
         self.session = session
-
         if self.session:
             self.remote_runner = remote.RemoteRunner(session=self.session)
             runner = self.remote_runner.run
         else:
             runner = process.run
 
-        if LIBVIRTD is None:
+        self.service_name = service_name
+        if self.service_name is None:
             logging.warning("Libvirtd service is not available in host, "
                             "utils_libvirtd module will not function normally")
-        self.libvirtd = service.Factory.create_service(LIBVIRTD, run=runner)
+        if libvirtd_decorator.get_libvirt_version_compare(5, 6, 0):
+            if libvirtd_decorator.get_libvirtd_split_enable_bit():
+                self.service_name = re.sub("libvirtd", "virtqemud",
+                                           self.service_name)
+            else:
+                self.service_name = re.sub("^virt.*d", "libvirtd",
+                                           self.service_name)
+        else:
```

Comment:
I think no need for this else

Reply:
No, it's necessary. The daemon_name can be libvirtd.socket or virtproxyd-tls.socket, which does not exist in lower version of libvirt(eg. libvirt 4.5.0).

Reply:
I mean the default value is LIBVIRTD which is 'libvirtd' by default, so may not need to set

Reply:
I've update the default value of service_name to None, please help check again.